app/explainer/agent_explainer.py

```
@@ -333,6 +334,113 @@ def _get_mock_iam_findings(self) -> List[SecurityFinding]:
             ),
         ]
 
+    def _analyze_with_context(
+        self, infra_findings: List[Dict], app_findings: List[Dict]
+    ) -> List[SecurityFinding]:
+        """Perform enhanced analysis with project context"""
+        if self.use_mock:
+            return self._get_enhanced_mock_findings()
+
+        prompt = build_analysis_prompt(infra_findings, app_findings, self.project_context)
```

Flake8 (7.2.0)

[error] undefined name 'build_analysis_prompt'

(F821)